UI/cypress/support/commands.js

```diff
@@ -24,19 +24,30 @@
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
 
-import { removeNewLine } from "./utils";
-
-import { getJumpToLocationId } from "./utils";
+import { removeNewLine, getJumpToLocationId } from "./utils";
 
 Cypress.Commands.overwrite("visit", (originalFn, url, options) => {
-  const username = Cypress.env("username");
-  const password = Cypress.env("password");
+  const basicAuthCredentials = Cypress.env("BASIC_AUTH_CREDENTIALS");
+
+  if (basicAuthCredentials) {
+    const [username, password] = basicAuthCredentials.split(":");
+    const auth = `${username}:${password}`;
+    const authHeader = `Basic ${btoa(auth)}`;
+
+    // Preserve the authentication header across multiple cy.visit() calls
+    Cypress.on("window:before:load", (win) => {
+      win.document.cookie = "auth=" + authHeader;
+    });
 
-  if (username && password) {
-    options = options || {};
-    options.auth = {
-      username: username,
-      password: password,
+    options = {
+      ...options,
+      headers: {
```

Comment:
Not sure why you're managing the headers yourself. Cypress already handles this for you if you set the `auth` option. https://docs.cypress.io/api/commands/visit#Add-basic-auth-headers